src/lib/incentive-relationship-calculation.ts

```diff
@@ -93,25 +122,50 @@ export function buildRelationshipGraph(data: IncentiveRelationships) {
   return edges;
 }
 
-// Checks the prerequisites for a single incentive.
-export function meetsPrerequisites(
+// Helper to traverse nested prerequisites and check whether they are met for
+// the given incentive.
+function meetsSinglePrerequisite(
   incentiveId: string,
+  prerequisite: IncentivePrerequisites,
   maps: RelationshipMaps,
 ) {
-  const prerequisiteIds = maps.requiresMap.get(incentiveId);
-  if (prerequisiteIds !== undefined) {
-    for (const prerequisiteId of prerequisiteIds) {
-      if (
-        !maps.eligibleIncentives.has(prerequisiteId) &&
-        maps.eligibleIncentives.has(incentiveId)
-      ) {
+  if (typeof prerequisite === 'string') {
+    if (
+      !maps.eligibleIncentives.has(prerequisite) &&
+      maps.eligibleIncentives.has(incentiveId)
+    ) {
+      return false;
+    }
+  } else if ('anyOf' in prerequisite) {
+    for (const child of prerequisite.anyOf) {
+      if (meetsSinglePrerequisite(incentiveId, child, maps)) {
+        return true;
+      }
+    }
+    return false;
+  } else if ('allOf' in prerequisite) {
```

Comment:
nit: now that there are more conditional branches, IMO it's clearer to have the return within each of the branches, even if it's a bit more verbose. Then each branch is self-contained rather than some (and only) some of the branches falling through.

Tangentially related: if the prerequisite is an unexpected case (i.e. not a string, 'anyOf', or 'allOf'), would we rather return true or false? It feels like false is more conservative.

Reply:
I didn't think the schema would allow that so I didn't think about it too hard. I think that would be some kind of weird error state so false seems fine with me.

I rewrote this to be a bit more verbose but probably clearer, let me know what you think.

Reply:
Yeah, I'm fine with relying on the test-time schema validation to prevent unexpected prerequisite shapes from appearing at runtime.